sdks/python/apache_beam/io/gcp/bigtable_io_test.py

```diff
@@ -0,0 +1,174 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Unittest for GCP Bigtable testing."""
+from __future__ import absolute_import
+
+import datetime
+import logging
+import random
+import string
+import unittest
+import uuid
+
+import apache_beam as beam
+from apache_beam.io.gcp.bigtable_io_write import BigtableWriteConfiguration
+from apache_beam.io.gcp.bigtable_io_write import WriteToBigtable
+from apache_beam.metrics.metric import MetricsFilter
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.runners.runner import PipelineState
+from apache_beam.testing.test_pipeline import TestPipeline
+
+# Protect against environments where bigtable library is not available.
+# pylint: disable=wrong-import-order, wrong-import-position
+try:
+  from google.cloud.bigtable import row, column_family, Client
+except ImportError:
+  Client = None
+
+
+class GenerateDirectRows(beam.DoFn):
+  """ Generates an iterator of DirectRow object to process on beam pipeline.
+
+  """
+
+  def process(self, row_values):
+    """ Process beam pipeline using an element.
+
+    :type row_value: dict
+    :param row_value: dict: dict values with row_key and row_content having
+    family, column_id and value of row.
+    """
+    direct_row = row.DirectRow(row_key=row_values["row_key"])
+
+    for row_value in row_values["row_content"]:
+      direct_row.set_cell(row_value["column_family_id"],
+                          row_value["column_id"],
+                          row_value["value"],
+                          datetime.datetime.now())
+      yield direct_row
```

Comment:
The yield should only happen once.

Reply:
Please pass in just an index into process, and call `_generate_mutation_data`, which should return a full `DirectRow`.  I think that would be a bit more efficient.